arm/Microsoft.ServiceBus/namespaces/queues/deploy.bicep

```diff
@@ -123,33 +123,39 @@ resource serviceBusNamespaceQueue 'Microsoft.ServiceBus/namespaces/queues@2021-0
     enablePartitioning: enablePartitioning
     enableExpress: enableExpress
   }
+}
 
-  resource serviceBusNamespaceQueue_authorizationRules 'authorizationRules@2017-04-01' = [for authorizationRule in authorizationRules: {
+module queue_authorizationRules 'authorizationRules/deploy.bicep' = [for (authorizationRule, index) in authorizationRules: {
+  name: '${deployment().name}-AuthRule-${index}'
+  params: {
+    namespaceName: namespaceName
+    queueName: last(split(queue.name, '/'))
     name: authorizationRule.name
-    properties: {
-      rights: authorizationRule.properties.rights
-    }
-  }]
-}
+    rights: contains(authorizationRule, 'rights') ? authorizationRule.rights : []
+  }
+  dependsOn: [
+    queue
+  ]
+}]
 
-resource serviceBusNamespaceQueue_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lock != 'NotSpecified') {
-  name: '${split(serviceBusNamespaceQueue.name, '/')[1]}-${lock}-lock'
+resource queue_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lock != 'NotSpecified') {
+  name: '${split(queue.name, '/')[1]}-${lock}-lock'
   properties: {
     level: lock
     notes: (lock == 'CanNotDelete') ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
   }
-  scope: serviceBusNamespaceQueue
+  scope: queue
 }
 
-module serviceBusNamespaceQueue_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+module queue_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
   name: '${deployment().name}-rbac-${index}'
   params: {
     roleAssignmentObj: roleAssignment
     builtInRoleNames: builtInRoleNames
-    resourceName: serviceBusNamespaceQueue.name
+    resourceName: queue.name
   }
 }]
 
-output namespaceQueueName string = serviceBusNamespaceQueue.name
-output namespaceQueueResourceId string = serviceBusNamespaceQueue.id
+output namespaceQueueName string = queue.name
+output namespaceQueueResourceId string = queue.id
```

Comment:
Dont have to fix it here, but are we going to have outputs on all modules? Want to fix that here or a separate Issue/PR?

Reply:
fixed